yarn-project/types/src/l2_block.ts

```diff
@@ -431,6 +432,22 @@ export class L2Block {
     });
   }
 
+  /**
+   * Helper function to attach encrypted logs related to a block. Since we can have L2 blocks without encrypted logs,
+   * this function helps attach them in order to make the block data manipulation easier.
+   * @param encryptedLogs - The encrypted logs to be attached to the block.
+   */
+  attachEncryptedLogs(encryptedLogs: NoirLogs) {
+    // throw error if the block already has encrypted logs attached.
```

Comment:
This doesn't throw an error but only returns. I think it would be a good idea to throw an error here.